js/geo/transform.js

```diff
@@ -408,6 +408,9 @@ class Transform {
         mat4.perspective(m, 2 * Math.atan((this.height / 2) / this.altitude), this.width / this.height, 0.1, farZ);
         mat4.translate(m, m, [0, 0, -this.altitude]);
 
+        // a hack around https://github.com/mapbox/mapbox-gl-js/issues/2270
+        m[14] = Math.min(m[14], m[15]);
```

Comment:
The rewards of this not-100%-understood fix outweigh the severity of the bug & risks of the fix in my mind.

Out of curiosity, is there any intuitive meaning of elements 14 and 15 in the transformation matrix? Can we anticipate any cases where elements 14 and 15 might end up with very different values? cc @ansis